src/ucp/core/ucp_ep.c

```diff
@@ -2623,6 +2623,24 @@ int ucp_ep_is_cm_local_connected(ucp_ep_h ep)
            (ep->flags & UCP_EP_FLAG_LOCAL_CONNECTED);
 }
 
+int ucp_ep_is_local_connected(ucp_ep_h ep)
+{
+    int is_local_connected = !!(ep->flags & UCP_EP_FLAG_LOCAL_CONNECTED);
+    ucp_wireup_ep_t *wireup_ep;
+    ucp_lane_index_t i;
+
+    if (ucp_ep_has_cm_lane(ep)) {
```

Comment:
1. so the issue is that in case of CM, we set UCP_EP_FLAG_LOCAL_CONNECTED before the lanes are connected?
 maybe add a comment, and/or check each lane also in no-cm case (and add assertions)?
2. maybe arrange code like this:
```
if (has_cm) {
 return 1;
}
for (lane = 0; lane < num_lanes; ++lane) {
 if (wireup_ep!=NULL && !(wireup_ep->flags &LOCAL_CONNECTED) {
 return 0;
}

return 1;l
```

Reply:
1. yes, it's used in close protocol. it's documented in `UCP_EP_FLAG_LOCAL_CONNECTED` description
2. IMHO if rewrite the code this way correctly, it becomes more complex

Reply:
1. i think it's worth to add comment in this function to explain why need to check all lanes